lib/index.js

```diff
@@ -56,84 +74,68 @@ module.exports = {
-        const { Key } = getPathKey(file, config.pool);
-        S3.upload(
-          {
+    const upload = async (file, customParams = {}) => {
+      const { Key } = getPathKey(file, config.pool);
+      try {
+        await S3.send(
+          new PutObjectCommand({
+            Bucket: config.params.Bucket,
+            ACL: config.params.ACL,
             Key,
             Body: file.stream || Buffer.from(file.buffer, "binary"),
             ContentType: file.mime,
             ...customParams,
-          },
-          (err, data) => {
-            if (err) {
-              return reject(err);
-            }
-
-            // When a file is >5MB, Location is set to "auto" and the bucket name is prepended to the key.
-
-            // Strip bucket name from key if Location is auto
-            const key =
-              data.Location === "auto" &&
-              data.Key.startsWith(`${config.params.Bucket}/`)
-                ? data.Key.replace(`${config.params.Bucket}/`, "")
-                : data.Key;
-
-            // Set the bucket file URL.
-            // If there is a custom endpoint for data access set, replace the upload endpoint with the read enpoint URL.
-            // Otherwise, use location returned from S3 API if it's not "auto"
-            if (config.cloudflarePublicAccessUrl) {
-              file.url =
-                config.cloudflarePublicAccessUrl.replace(/\/$/g, "") +
-                "/" +
-                key;
-            } else if (data.Location !== "auto") {
-              file.url = data.Location;
-            } else {
-              throw new Error(
-                "Cloudflare S3 API returned no file location and cloudflarePublicAccessUrl is not set. strapi-provider-cloudflare-r2 requires cloudflarePublicAccessUrl to upload files larger than 5MB. https://github.com/trieb-work/strapi-provider-cloudflare-r2#provider-configuration"
-              );
-            }
-
-            // check if https is included in file URL
-            if (!assertUrlProtocol(file.url)) {
-              // Default protocol to https protocol
-              file.url = `https://${file.url}`;
-            }
-
-            resolve();
-          }
+          })
         );
-      });
 
-    return {
-      uploadStream(file, customParams = {}) {
-        return upload(file, customParams);
-      },
-      upload(file, customParams = {}) {
-        return upload(file, customParams);
-      },
-      delete(file, customParams = {}) {
-        return new Promise((resolve, reject) => {
-          // delete file on S3 bucket
-          const { Key } = getPathKey(file, config.upload);
-          S3.deleteObject(
-            {
-              Key,
-              ...customParams,
-            },
-            (err, data) => {
-              if (err) {
-                return reject(err);
-              }
-
-              resolve();
-            }
-          );
-        });
-      },
+        file.url = getFileURL(file, config);
```

Comment:
This should not assign back into `file.url`, the getFileURL is already doing that and causes this to overwrite and make it null, causing rendering issues.